services/app-web/src/common-code/ContractType.ts

```diff
@@ -1,36 +1,37 @@
 import { Contract, ContractRevision } from '../gen/gqlClient'
+import { getLastContractSubmission } from '../gqlHelpers/contractsAndRates'
 
-const getContractRev = (contract: Contract): ContractRevision => {
+const getContractRev = (contract: Contract): ContractRevision | undefined => {
     if (contract.draftRevision) {
         return contract.draftRevision
     } else {
-        return contract.packageSubmissions[0].contractRevision
+        return getLastContractSubmission(contract)?.contractRevision
     }
 }
 const isContractOnly = (contract: Contract): boolean => {
```

Comment:
No change needed now just sharing thought when I see all these `?` introduced to our utility functions.  

Have all these optional checks around the contract revision's existence means the boolean because less sure - we could be getting `false` for mal-formatted data, not just for it not being a good contract. I like seeing utilities, especially on the frontend like this (which literally determining display state) be more strictly typed. We would have to handle data validation/ correctness checks much earlier in the React component, divert to an error state if theres any question about the data shape.  Also starting to wonder if helpers should take the `ContractFormData` not the `Contract` along those lines - maybe have the component itself figure out which form data to pass? Not sure.